guide/src/conversions.md

```diff
@@ -1,8 +1,103 @@
 # Type Conversions
 
+In this portion of the guide we'll talk about the mapping of Python types to Rust types offered by PyO3, as well as the traits available to perform conversions between them.
+
+## Mapping of Rust types to Python types
+
+When writing functions callable from Python (such as `#[pyfunction]`s and `#[pymethods]`) consult the tables below to determine the correct Rust types to use for argument and return types.
+
+### Argument Types
+
+When you accept a function argument, you can choose to either use Rust library types or PyO3's Python-native types. (See the next section for discussion on when to use each.)
+
+The table below contains the Python type and the corresponding Rust library type or PyO3 Python-native type that can be used as a function argument:
+
+| Python        | Rust                            | Rust (Python-native) |
+| ------------- |:-------------------------------:|:--------------------:|
+| `Any`         | -                               | `&PyAny`             |
+| `str`         | `String`, `&str`, `Cow<str>`    | `&PyUnicode`         |
+| `bytes`       | `&[u8]`                         | `&PyBytes`           |
+| `bool`        | `bool`                          | `&PyBool`            |
+| `int`         | Any integer type (`i32`, `u32`, `usize`, etc) | `&PyLong` |
+| `float`       | `f32`, `f64`                    | `&PyFloat`           |
+| `complex`     | `num_complex::Complex`[^1]      | `&PyComplex`         |
+| `Optional[T]` | `Option<T>`                     | -                    |
+| `List[T]`     | `Vec<T>`                        | `&PyList`            |
+| `Dict[K, V]`  | `HashMap<K, V>`, `BTreeMap<K, V>` | `&PyDict`          |
+| `Tuple[T, U]` | `(T, U)`                        | `&PyTuple`           |
```

Comment:
`Vec<T>` is also works (important for instances in Python where tuples are used as "frozenlist" instead of struct-like tuples).

Reply:
👍